tools/opensnoop.py

```diff
@@ -453,7 +491,12 @@ def print_event(cpu, data, size):
                     event.comm, fd_s, err), nl="")
 
             if args.extended_fields:
-                printb(b"%08o " % event.flags, nl="")
+                # If neither O_CREAT nor O_TMPFILE is specified in flags, then
+                # mode is ignored, see open(2).
+                if event.mode == 0:
```

Comment:
> but when it is not a file creation case, 0000 represents the default value of the mode variable.

This is the only case where I suggested using a distinct expression.
If you handle all the zero values this way, won't it fail to meet your requirements below?

> When a program creates a file with mode=0000, it cannot even access the file
itself. It would be helpful if we could track the mode value. so we can know
who did it.

Reply:
Thanks, please review!